homeassistant/components/simplisafe/__init__.py

```diff
@@ -191,6 +203,58 @@
 CONFIG_SCHEMA = cv.deprecated(DOMAIN)
 
 
+@callback
+def _async_get_system_for_service_call(
+    hass: HomeAssistant, call: ServiceCall
+) -> SystemType:
+    """Get the SimpliSafe system related to a service call (by device ID)."""
+    device_id = call.data[ATTR_DEVICE_ID]
+    device_registry = dr.async_get(hass)
+
+    if (
+        alarm_control_panel_device_entry := device_registry.async_get(device_id)
+    ) is None:
+        raise vol.Invalid("Invalid device ID specified")
+
+    if TYPE_CHECKING:
+        assert alarm_control_panel_device_entry.via_device_id
+
+    if (
+        base_station_device_entry := device_registry.async_get(
+            alarm_control_panel_device_entry.via_device_id
+        )
+    ) is None:
+        raise ValueError("No base station registered for alarm control panel")
+
+    [system_id] = [
```

Comment:
Are the brackets here intended?

Reply:
Roger. The list comprehension will return a list with a single value, so the brackets will unpack that directly into the variable.

Reply:
👍 That's a interesting syntax quirk.